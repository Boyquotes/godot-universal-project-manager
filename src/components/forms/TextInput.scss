.text-input {
  background-color: $background-tint;
  border: 1px solid $accent-main;
  border-radius: 5px;
  color: $text-light;
  font-size: 1.2rem;
  padding: 10px;

  display: flex;
  flex-direction: column;
}


.text-input input {
  border-radius: 5px;
  background-color: $background-main;
  border: 1px solid $accent-main;
  color: $text-light;
  font-size: 1.2rem;
  padding: 10px;
}


.text-input input:focus {
  outline: none;
  border: 1px solid $accent-hover;
}