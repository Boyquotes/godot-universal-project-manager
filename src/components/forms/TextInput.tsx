interface props {
  label: string
  inputId: string
}


export default function TextInput({label, inputId}: props) {
  return (
    <div className="text-input">
      <label htmlFor={inputId}>{label}</label>
      <input type="text" id={inputId} defaultValue={''}></input>
    </div>
  )
}