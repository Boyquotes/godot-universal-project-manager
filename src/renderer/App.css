@import url('https://fonts.googleapis.com/css2?family=Quicksand&display=swap');

@font-face {
  font-family: "Light";
  /*Can be any text*/
  src: local("DroidSans"),
    url("./fonts/DroidSans.ttf") format("truetype");
}



body {
  font-family: 'Quicksand', sans-serif;
  background-color: #0C0910;
  color: #FFF8F0;
}


header {
  margin: 0;
  padding: 0;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;

  background-color: #3E3944;

  border-bottom: 2px solid #63B995;
}

header h3 {
  margin: 5px;
}

.header-top {
  margin-right: 5px;
  margin-left: 5px;
  display: flex;
  flex-direction: row;
  align-items: center;

  justify-content: space-between;
}

.header-top a {
  color: #AD5D4E;
  transition: all 0.2s ease-out;
}

.header-top a:hover {
  color: #FFF8F0;
}

.header-tabs {
  display: flex;
  flex-direction: row;
  justify-content: left;
  align-items: center;

  margin: 0px;
  padding: 0px;
}

.header-tab {
  transition: all 0.2s ease-out;

  padding-left: 10px;
  padding-right: 10px;

  min-width: 120px;

  margin-top: 5px;

  color: #AD5D4E;
  text-align: center;
  text-decoration: none;

  margin-left: 0.5px;
  margin-right: 0.5px;

  background-color: #272932;

  border: 2px solid #63B995;
  border-bottom-style: none;
  border-radius: 20px 20px 0px 0px;
}

.header-tab:hover {
  color: #FFF8F0;
  padding-top: 5px;
  margin-top: 0px;
}
