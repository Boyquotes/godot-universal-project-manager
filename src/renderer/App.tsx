import { MemoryRouter as Router, Routes, Route } from 'react-router-dom';
import icon from '../../assets/icon.svg';

import './App.css';
import './fonts/DroidSans.ttf';

import Layout from './Layout';
import NoPage from './pages/NoPage';
import Versions from './pages/Versions';
import Projects from './pages/Projects';
import Config from './pages/Config';

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Projects />} />
          <Route path="/versions" element={<Versions />} />
          <Route path="/config" element={<Config />} />
          <Route path="*" element={<NoPage />} />
        </Route>
      </Routes>
    </Router>
  );
}
