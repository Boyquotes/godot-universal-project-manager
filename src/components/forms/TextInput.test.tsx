import renderer from 'react-test-renderer';
import TextInput from './TextInput';


/**
 * @jest-environment jsdom
 */



describe('renders without crashing', () => {
  test('Component is rendered without breaking changes', () => {
    const component = renderer.create(
      <TextInput label="Test Label" inputId=""/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});