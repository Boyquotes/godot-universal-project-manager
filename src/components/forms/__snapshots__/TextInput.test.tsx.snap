// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing Component is rendered without breaking changes 1`] = `
<div
  className="text-input"
>
  <label
    htmlFor=""
  >
    Test Label
  </label>
  <input
    defaultValue=""
    id=""
    type="text"
  />
</div>
`;
